src/coreclr/jit/lower.cpp

```diff
@@ -7752,6 +7777,66 @@ GenTree* Lowering::LowerSignedDivOrMod(GenTree* node)
     return node->gtNext;
 }
 
+//------------------------------------------------------------------------
+// TryFoldBinop: Try removing a binop node by constant folding.
+//
+// Parameters:
+//   node - the node
+//
+// Returns:
+//   True if the node was removed
+//
+bool Lowering::TryFoldBinop(GenTreeOp* node)
+{
+    if (node->gtSetFlags())
+    {
+        return false;
+    }
+
+    GenTree* op1 = node->gtGetOp1();
+    GenTree* op2 = node->gtGetOp2();
+
+    if (op1->IsIntegralConst() && op2->IsIntegralConst() &&
+        op1->AsIntConCommon()->ImmedValCanBeFolded(comp, node->gtOper) &&
+        op2->AsIntConCommon()->ImmedValCanBeFolded(comp, node->gtOper))
+    {
+        GenTree* folded = comp->gtFoldExprConst(node);
+        assert(folded == node);
+        if (!folded->OperIsConst())
+        {
+            return false;
+        }
+
+        op1->SetUnusedValue();
+        op2->SetUnusedValue();
```

Comment:
Do we often just mark nodes as unused without deleting them?

Reply:
Seems better to just remove them given they are simple constants that won't ever have side effects. Changed to that.